### PR TITLE
[MBL-15793][Student] Dashboard Invitation interaction tests

### PR DESCRIPTION
**Pull request by hermannakos**
refs: MBL-15793
affects: Student
release note: none

test plan: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/student/src/androidTest/java/com/instructure/student/ui/interaction/DashboardInteractionTest.kt [hunk 1]
```diff
@@ -163,6 +165,36 @@ class DashboardInteractionTest : StudentTest() {
...
+        val data = getToDashboard(courseCount = 2, invitedCourseCount = 1)
+        val invitedCourse = data.courses.values.first { it.enrollments?.any { it.enrollmentState == EnrollmentAPI.STATE_INVITED } ?: false }
+
+        dashboardPage.assertInviteShowing(invitedCourse.name)
+        dashboardPage.refresh() //need this refresh because if there are such amount of elements and the screen is scrollable, first "interaction" will scroll down somehow a bit. It works on physical device, it's just an emulator-specific issue.
+        dashboardPage.assertInviteShowing(invitedCourse.name)
+        dashboardPage.acceptInvite()
+        dashboardPage.assertInviteAccepted()
+        dashboardPage.refresh()
+        dashboardPage.assertInviteGone(invitedCourse.name)
```
Comment (kdeakinstructure) [inline apps/student/src/androidTest/java/com/instructure/student/ui/interaction/DashboardInteractionTest.kt:180]: I'd assert that the course is displayed on the courses list as well, just to make sure that the invitation accept does it's job.

File: apps/student/src/androidTest/java/com/instructure/student/ui/interaction/DashboardInteractionTest.kt [hunk 2]
```diff
@@ -163,6 +165,36 @@ class DashboardInteractionTest : StudentTest() {
...
+        val data = getToDashboard(courseCount = 2, invitedCourseCount = 1)
+        val invitedCourse = data.courses.values.first { it.enrollments?.any { it.enrollmentState == EnrollmentAPI.STATE_INVITED } ?: false }
+
+        dashboardPage.assertInviteShowing(invitedCourse.name)
+        dashboardPage.refresh() //need this refresh because if there are such amount of elements and the screen is scrollable, first "interaction" will scroll down somehow a bit. It works on physical device, it's just an emulator-specific issue.
+        dashboardPage.assertInviteShowing(invitedCourse.name)
+        dashboardPage.declineInvite()
+        dashboardPage.assertInviteDeclined()
+        dashboardPage.refresh()
+        dashboardPage.assertInviteGone(invitedCourse.name)
```
Comment (kdeakinstructure) [inline apps/student/src/androidTest/java/com/instructure/student/ui/interaction/DashboardInteractionTest.kt:195]: Same as above. (To check that after decline, the course does not displayed on dashboard page)

**Review by kdeakinstructure — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]

**Pull request merged**